raster/r.proj/testsuite/test_rproj.py

```diff
@@ -0,0 +1,183 @@
+"""
+Name:       r.texture test
+Purpose:    Tests r.texture and its flags/options.
+
+Author:     Sunveer Singh, Google Code-in 2017
+            Chung-Yuan Liang, modified in 2024
+Copyright:  (C) 2017 by Sunveer Singh and the GRASS Development Team
+Licence:    This program is free software under the GNU General Public
+            License (>=v2). Read the file COPYING that comes with GRASS
+            for details.
+"""
+
+from grass.gunittest.case import TestCase
+from grass.gunittest.checkers import text_to_keyvalue
+from grass.gunittest.gmodules import call_module
+import os
+
+raster_info = """north=35.8096296297222
+south=35.6874074075
+east=-78.608
+west=-78.7746666666667
+nsres=0.000740740740740727
+ewres=0.000666666666666686
+rows=165
+cols=250
+cells=41250"""
+
+src_project = "nc_spm_full_v2alpha2"
+dst_project = "nc_latlong"
+
+
+class TestRasterreport(TestCase):
+    input = "elevation"
+
+    @classmethod
+    def setUpClass(cls):
+        cls.runModule("g.proj", project=dst_project, epsg="4326", flags="c")
+        cls.runModule("g.mapset", mapset="PERMANENT", project=dst_project)
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.runModule("g.mapset", mapset="PERMANENT", project=src_project)
+        dbase = call_module("g.gisenv", get="GISDBASE")
+        os.system(f"rm -rf {dbase}/{dst_project}")
+
+    def run_rproj_test(self, method, statics):
+        """The main function to run r.proj check rsults according to the method
+
+        Parameters
+        ----------
+        method : str
+            The method to be used for r.proj
+        statics : str
+            The expected statics of the output raster
+        """
+        output = method
+        ## Get the boundary and set up region for the projected map
+        stdout = call_module(
+            "r.proj",
+            project=src_project,
+            mapset="PERMANENT",
+            input=self.input,
+            method=method,
+            flags="g",
+        )
+        settings = dict([line.split("=") for line in stdout.split()])
+
+        call_module(
+            "g.region",
+            n=settings["n"],
+            s=settings["s"],
+            e=settings["e"],
+            w=settings["w"],
+            rows=settings["rows"],
+            cols=settings["cols"],
+            flags="a",
+            res=1,
+        )
+
+        ## Project the map
+        self.assertModule(
+            "r.proj",
+            project=src_project,
+            mapset="PERMANENT",
+            input=self.input,
+            output=output,
+            method=method,
+            quiet=True,
+        )
+
+        ## Validate the output
+        self.assertRasterFitsUnivar(output, reference=statics, precision=1e-2)
+        self.assertRasterFitsInfo(output, reference=raster_info, precision=1e-2)
```

Comment:
Was that choice of precision based on something existing? It seems a bit large for me, possibly hiding some architectural changes, like the ones we discovered when building for macOS on aarch64. We still don't don't if the cause of the problem is aarch64 or M1+/macOS, as we didn't test other aarch64 on Linux.

https://github.com/OSGeo/grass/issues/3398 https://github.com/OSGeo/grass/issues/3397

Reply:
No, I tried using 1e-7, and the tests work fine on my arm-based Mac. It was just because I forgot to change the number.